raiden-ts/src/transfers/epics/mediate.ts

```diff
@@ -34,34 +79,51 @@ function shouldMediate(action: transferSigned, address: Address, { caps }: Raide
  * @param deps - Raiden epic deps
  * @param deps.address - Our address
  * @param deps.config$ - Latest observable
+ * @param deps.log - Logger instance
  * @returns Observable of outbound transfer.request actions
  */
 export function transferMediateEpic(
   action$: Observable<RaidenAction>,
-  {}: Observable<RaidenState>,
-  { address, config$ }: RaidenEpicDeps,
+  state$: Observable<RaidenState>,
+  { address, config$, log }: RaidenEpicDeps,
 ) {
   return action$.pipe(
     filter(transferSigned.is),
-    withLatestFrom(config$),
+    withLatestFrom(config$, state$),
     filter(([action, config]) => shouldMediate(action, address, config)),
-    map(([{ payload: { message: transf }, meta: { secrethash } }]) =>
+    map(([action, config, state]) => {
+      const transferState = state.transfers[transferKey(action.meta)];
+      const locked = action.payload.message;
+
+      const tokenNetwork = locked.token_network_address;
+      const secrethash = action.meta.secrethash;
+      const receivedPartner = transferState.partner;
+
+      const routes = filterRoutes(locked.metadata.routes, state, tokenNetwork, { address, log });
+      if (!routes) return;
```

Comment:
Please, refer to the var type. `route` is only falsy (`null`) if no valid forward routes were found, and there was a log call on this case. Since this was on a helper function, this is how we forward it to the external world, and the consequence is that we just drop this transfer if we can't mediate it, and let it expire (this is actually PC's implementation, and it defines the intended behavior per Raiden protocol of reference implementation).